### PR TITLE
Widgets and Listeners class structure.

### PR DESCRIPTION
**Pull request by lettucewrap**
Widgets and Listeners class structure.

### CODE REVIEW

File: src/main/java/version1/gameUtil/widgets/buttons/MazeRegisterButton.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package version1.gameUtil.widgets.buttons;
+
+import version1.gameUtil.listeners.interfaces.LoginBListenerIF;
+
+import java.awt.event.ActionListener;
+
+public class MazeRegisterButton extends MazeButton{
+    private LoginBListenerIF mediator;
```
Comment (BenMob) [inline src/main/java/version1/gameUtil/widgets/buttons/MazeRegisterButton.java]: Should the declared type here be RegistrationBListenerIF ?

File: src/main/java/version1/gameUtil/widgets/inputfields/MazeTextField.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package version1.gameUtil.widgets.inputfields;
+
+import version1.gameUtil.listeners.implementations.AuthenticationManager;
+import version1.gameUtil.listeners.interfaces.LoginBListenerIF;
+
+public class MazeTextField extends AbstractMazeTextField {
+    private LoginBListenerIF mediator;
```
Comment (BenMob) [inline src/main/java/version1/gameUtil/widgets/inputfields/MazeTextField.java]: Same here, should the declared type be MazeTextFieldIF ?

**Review by BenMob — CHANGES_REQUESTED** [review]
Other than the two comments, I think everything looks great!

**Review by BenMob — APPROVED** [review]

File: src/main/java/version1/gameUtil/widgets/inputfields/MazeTextField.java (hunk 2 above)
Comment (lettucewrap) [reply src/main/java/version1/gameUtil/widgets/inputfields/MazeTextField.java]: Got it, thanks!

**Pull request merged**